### DISCUSSION TITLE
Changing the request method results in missing value error

### DISCUSSION THREAD

**Discussion by vishal-ramesh2** [fastapi/fastapi#6409] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.post(
    "/api/{entity_id}/somepath/bulk-map",
    response_model=CustomResponseModel,
    status_code=status.HTTP_201_CREATED,
    response_model_exclude_none=True,)
def map_entity_to_data(
    background_tasks: BackgroundTasks,
    entity_id: str,
    map_request: CustomRequestModel = Body(
        ...,
        example={
            "response": {"sample": {"id": "121212", "order_no": 1, "name": "test"}},
            "data_ids": ["708fbb80a3239167c19891d2", "708fbb80a3239167c19891d2"],
        },
    ),
    user_id: str = Header(..., convert_underscores=False),
) -> CustomResponseModel:
    //Some processing
```

### Description

I have an API with an endpoint defined similar to the one shown in the example. This API works fine as it is. But when I change the request methoed from POST to PUT (i.e., `router.post` to `router.put`), I get the error
```
{'errors': [{'loc': ['body', 'data'], 'msg': 'field required', 'type': 'value_error.missing'}]}
```

I get this error when I use swagger (generated by fastapi) and also when I make the request using `requests`

I don't have a field named `data` and I don't understand why changed it to put is causing this issue.

### Operating System

macOS

### FastAPI Version

0.78.0

### Python Version

3.8.5

**Comment by tiangolo** [comment, marked as answer]
I imagine you solved your problem, so thanks for coming back to close the issue! 🍰 

> Sorry for the long delay! 🙈  I wanted to personally address each issue and they piled up through time, but now I'm checking each one in order.

**Reply by vishal-ramesh2** [reply]
Sorry I forgot to close this. I think I found the issue back then. It was happening because I had conflicting endpoints if I remember correctly so the payload was trying to match with another endpoint.
